.github/workflows/publish-containers.yml

```diff
@@ -0,0 +1,113 @@
+name: Publish docker image
+
+on:
+  push:
+    tags:
+      # Matches stable and pre-releases
+      - "[0-9]+.[0-9]+.[0-9]+*"
+    branches:
+      - main
+      - ci/**
+
+jobs:
+  docker:
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      # Outputs the name of the repository (owner/repo)
+      - name: Get image name
+        id: image
+        run: |
+          IMAGE_NAME=$(echo ${{ github.repository }} | tr '[:upper:]' '[:lower:]')
+          echo ::set-output name=image_name::ghcr.io/${IMAGE_NAME}
+
+      # Tags stable versions as :latest
+      # Pre-releases, alphas, etc. as :snapshot
+      - name: Output image tags from git tag events
+        if: ${{ startsWith(github.ref, 'refs/tags/') }}
+        run: |
+          set -x
+          # Remove everything else than the tagged version
+          VERSION=${GITHUB_REF#refs/tags/}
+          # Tag as stable (:latest) if there is no letters in the version
+          # Otherwise, tag as preview (:snapshot)
+          if [[ $VERSION =~ ^[0-9]+\.[0-9]+\.[0-9]+$ ]]; then
+            # Matches 1.0.0, 1.0.1, etc.
+            NAMED_VERSION="latest"
+          else
+            # 1.0.0a1, 1.0.0rc1, etc.
+            NAMED_VERSION="snapshot"
+          fi
+          TAGS=$"\
+          ${{ steps.image.outputs.image_name }}:${NAMED_VERSION},\
+          ${{ steps.image.outputs.image_name }}:${VERSION}\
+          "
+          echo "
+          CONTAINER_TAGS=${TAGS}
+          NAMED_VERSION=${NAMED_VERSION}
+          " >> "${GITHUB_ENV}"
```

Comment:
This workflow is copying the current known issues with core's CI

## Issue no.1
This assumes there is only one minor version maintained at a time, it's not true. I would recommend not to tag `latest` and not to tag `snapshot`, only tag the version

## Issue no.2
It doesn't support minor tags, only patch tags. When we push `1.2.3`, we should push `1.2` as well.